specification/dartLangSpec.tex

```diff
@@ -2126,22 +2126,25 @@ \section{Classes}
 are its static and instance members.
 
 \LMHash{}%
-A class has several scopes:
+A class declaration has two scopes:
 \begin{itemize}
-\item A \IndexCustom{type-parameter scope}{scope!type parameter},
+\item
+  A \IndexCustom{type-parameter scope}{scope!type parameter},
   which is empty if the class is not generic (\ref{generics}).
-The enclosing scope of the type-parameter scope of a class is the enclosing scope of the class declaration.
-\item A \IndexCustom{static scope}{scope!static}.
-The enclosing scope of the static scope of a class is the type parameter scope (\ref{generics}) of the class.
-\item An \IndexCustom{instance scope}{scope!instance}.
-The enclosing scope of a class' instance scope is the class' static scope.
+  The enclosing scope of the type-parameter scope of a class declaration is
+  the current scope of the class declaration
+  (\commentary{which is the library scope of the current library}).
+\item
+  A \IndexCustom{class scope}{scope!class}.
+  The enclosing scope of the class scope of a class declaration is
+  the type parameter scope of the class declaration.
 \end{itemize}
 
 \LMHash{}%
-The enclosing scope of an instance member declaration is the instance scope of the class in which it is declared.
-
-\LMHash{}%
-The enclosing scope of a static member declaration is the static scope of the class in which it is declared.
+The enclosing scope of an instance member declaration,
```

Comment:
We use "enclosing scope" about a scope, meaning the an *outer scope*, above, and here we use "enclosing scope" of a declaration to be the scope containing the declaration. That's confusing.

If we had done the same above, we could say
"The enclosing scope of the type-parameter scope of a class declaration is the enclosing scope of the class declaration" (which I first tried as an alternative to "current scope", and then found quite unreadable).

Could we use "enclosing scope" only for scopes (scopes are nested) and "containing scope" for declarations (each declaration belongs to exactly one scope)?

Reply:
That's a mistake, it should be 'The current scope'.

> Could we use "enclosing scope" only for scopes 

Yes, that's how it should be. Changed 13 locations (a few of them were obsolete comments that I just deleted) to make it so.

We may wish to change 'current scope' to some other phrase, but this at least makes the text more consistent such that we can change 'current scope' more reliably, if we choose to do that.

I did not change a couple of locations where it talks about 'enclosing scopes' (of a term). That's taken to mean 'the current scope of the term and its enclosing scopes'. We could change that, too, but I haven't done that.